elf_reader_test.go

```diff
@@ -290,29 +291,116 @@ func TestLibBPFCompat(t *testing.T) {
 		t.Skip("No path specified")
 	}
 
-	testutils.TestFiles(t, filepath.Join(*elfPath, *elfPattern), func(t *testing.T, path string) {
+	load := func(t *testing.T, spec *CollectionSpec, opts CollectionOptions, valid bool) {
+		// Disable retrying a program load with the log enabled, it leads
+		// to OOM kills.
+		opts.Programs.LogSize = -1
+		coll, err := NewCollectionWithOptions(spec, opts)
+		testutils.SkipIfNotSupported(t, err)
+		var errno syscall.Errno
+		if errors.As(err, &errno) {
+			// This error is most likely from a syscall and caused by us not
+			// replicating some fixups done in the selftests. This is expected,
+			// so skip the test instead of failing.
+			t.Skip("Skipping since the kernel rejected the program:", errno)
+		}
+		if err == nil {
+			coll.Close()
+		}
+		if !valid {
+			if err == nil {
+				t.Fatal("Expected an error during load")
+			}
+		} else if err != nil {
+			t.Fatal("Error during loading:", err)
+		}
+	}
+
+	files := testutils.Glob(t, filepath.Join(*elfPath, *elfPattern),
+		// These files are only used as a source of btf.
+		"btf__core_reloc_*",
+	)
+
+	testutils.Files(t, files, func(t *testing.T, path string) {
 		file := filepath.Base(path)
 		switch file {
-		case "test_ksyms.o":
-			// Issue #114
-			t.Skip("Kernel symbols not supported")
 		case "test_sk_assign.o":
-			t.Skip("Incompatible struct bpf_map_def")
-		case "test_ringbuf_multi.o", "map_ptr_kern.o", "test_btf_map_in_map.o":
-			// Issue #155
-			t.Skip("BTF .values not supported")
+			t.Skip("Skipping due to incompatible struct bpf_map_def")
```

Comment:
Note: this incompatible struct is actually called `struct bpf_elf_map`. I've investigated ways of supporting this legacy struct before, and I still think it might be worthwhile to at least be able to parse this struct successfully and exposing the trailing bits to the caller, if not just supporting it natively out of the box.

Reply:
On second thought, this might put on more code complexity than I initially assumed. We should fix this string, though. :point_up: